### PR TITLE
[ci] Remove `tune-sklearn` doc dependency

### PR DESCRIPTION
**Pull request by justinvyu**
## Why are these changes needed?

This unpins `tune-sklearn` as a dependency of the ray docs build. This updates any auto-generated API references with links to a self-contained docs page in the `tune-sklearn` repo.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by matthewdeng — APPROVED** [review]
(pending compiled requirements update)

<div><br/>via <a href="https://giphy.com/boomerangtoons/">Boomerang Official</a> on <a href="https://giphy.com/gifs/lVBtp4SRW6rvDHf1b6">GIPHY</a></div>

File: python/requirements_compiled.txt [hunk 1]
```diff
@@ -2390,7 +2390,6 @@ transformers==4.19.1
     # via -r /ray/ci/../python/requirements/ml/core-requirements.txt
 trustme==0.9.0
     # via -r /ray/ci/../python/requirements/test-requirements.txt
-tune-sklearn @ git+https://github.com/ray-project/tune-sklearn@master
```
Comment (aslonnie) [inline python/requirements_compiled.txt:2393]: why is this lost now?

Comment (justinvyu) [reply python/requirements_compiled.txt:2393]: I was confused about this as well, I just took the file from the CI artifact: https://buildkite.com/ray-project/premerge/builds/9981#018b6dab-1668-4452-87e1-8d923270a410

Comment (aslonnie) [reply python/requirements_compiled.txt:2387]: ah, it is removed.

**Review by aslonnie — APPROVED** [review]

**Pull request merged**